boards/px4/fmu-v6xrt/nuttx-config/scripts/script.ld

```diff
@@ -53,22 +53,36 @@ ENTRY(__start)
 
 SECTIONS
 {
-    /* Image Vector Table and Boot Data for booting from external flash */
+    /* Provide dummy relocation symbols for the dummy ToC */
+    PROVIDE(_toc_start = 0);
+    PROVIDE(_toc_end = 0);
+    PROVIDE(_toc_signature = 0);
+    PROVIDE(_app_start = 0);
+    PROVIDE(_app_end = 0);
+    PROVIDE(_boot_signature = 0);
+
+    /* Make a hole for the ToC and signature */
+
+    .toc (NOLOAD) : {
+        *(.main_toc)
+        *(.main_toc_sig)
+        FILL(0xff)
+        . = 0x400 ;
+    } > flash
 
     .boot_hdr : ALIGN(4)
     {
         FILL(0xff)
-        . = 0x400 ;
         __boot_hdr_start__ = ABSOLUTE(.) ;
         KEEP(*(.boot_hdr.conf))
-        . = 0x1000 ;
+        . = 0x1000 - 0x400 ;
```

Comment:
What is the reason for moving everything back by -0x400?

Reply:
Originally the first 1KB was just filled with FF. I added the TOC (size of 0x400) in there, keeping everything else in place. so this section now starts already at 0x400. The ". =" is relative to section start...

Reply:
I know how the location pointer works. So the ouput section was moved forward because you put the .toc output section prior to this.

What I assumed was, that the original code relies on this, i.e. boot_hdr.conf is at output section offset 0, and boot_hdr.ivt is at offset 1000h from this and so forth. Now the offsets are different.